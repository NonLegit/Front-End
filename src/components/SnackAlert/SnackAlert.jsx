import CloseIcon from '@mui/icons-material/Close';
import {
  Container, Close, Image, Info,
} from './styles';
/**
 * - Notifications Body
 * - fech notifications data from api
 *  @component
 * @param {Object} info - data of alert like message and sucess or fail
 * @return {React.Component} - Snack Alert
 */
function SnackAlert(info) {
  const message = info;
  console.log(message);
  return (
    <div style={message?.info.style}>
      <Container>
        <Close className="close" onClick={message?.info.close} type={message?.info.message?.type}>
          <CloseIcon className="icon" />
        </Close>
        <Info>
          <Image src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTWMfN2o-u82351CC0A0yOhqFvuCccfnTUA-Q&usqp=CAUs" />
          { message?.info.message?.text }
        </Info>
      </Container>
    </div>

  );
}

export default SnackAlert;
