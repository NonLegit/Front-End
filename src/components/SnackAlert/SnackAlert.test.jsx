import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import SnackAlert from './SnackAlert';
// test snapshot
test('test snapshot', async () => {
  const info = {

    info: {
      style: '',
      close: '',
      message: {
        type: '',
      },
    },
  };
  const tree = renderer.create(
    <Router>
      <SnackAlert info={info} />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
