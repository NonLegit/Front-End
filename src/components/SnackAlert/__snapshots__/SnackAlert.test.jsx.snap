// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div>
  <div
    className="css-mbgpr2"
  >
    <div
      className="close css-gwbri7"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      className="css-18p239j"
    >
      <img
        className="css-1koqu17"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTWMfN2o-u82351CC0A0yOhqFvuCccfnTUA-Q&usqp=CAUs"
      />
    </div>
  </div>
</div>
`;
